### PR TITLE
[SOAR-5536] InsightVM Tag Assets new action

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Add a new action that allows users to tag multiple assets in bulk

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [x] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [x] Screenshot of job output with the plugin changes
- [x] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

Screenshot 2022-05-19 at 12 21 45
Screenshot 2022-05-19 at 12 21 24

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [x] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [x] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [x] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [x] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [x] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/rapid7_insightvm/komand_rapid7_insightvm/util/endpoints.py [hunk 1]
```diff
@@ -35,6 +35,16 @@ def asset_tags(console_url, asset_id, tag_id=None):
         else:
             return f"{console_url}/api/3/assets/{asset_id}/tags"
 
+    @staticmethod
+    def tag_assets(console_url, tag_id):
```
Comment (igorski-r7) [inline plugins/rapid7_insightvm/komand_rapid7_insightvm/util/endpoints.py:39]: Great to see docstring here! 😃  Would you like to add type hints as well?

File: plugins/rapid7_insightvm/unit_test/test_tag_assets.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        assistance = (
+            "Verify your connection is pointing to your local console and not "
+            "`exposure-analytics.insight.rapid7.com` Ensure that the requested resource exists."
+        )
+        data = "The ID 2 is not available. Enter a different ID for this tag."
+        with self.assertRaises(PluginException) as e:
+            self.action.run(self.params)
+        self.assertEqual(cause, e.exception.cause)
+        self.assertEqual(assistance, e.exception.assistance)
+        self.assertEqual(data, e.exception.data)
```
Comment (igorski-r7) [inline plugins/rapid7_insightvm/unit_test/test_tag_assets.py:45]: Well personally I don't like single letter variables, but it's up to you 😃 
```suggestion
        with self.assertRaises(PluginException) as error:
            self.action.run(self.params)
        self.assertEqual(cause, error.exception.cause)
        self.assertEqual(assistance, error.exception.assistance)
        self.assertEqual(data, error.exception.data)
```

File: plugins/rapid7_insightvm/unit_test/test_tag_assets.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        assistance = (
+            "Verify your connection is pointing to your local console and not "
+            "`exposure-analytics.insight.rapid7.com` If this issue persists contact support for assistance."
+        )
+        data = "An unexpected error occurred. See the nsc.log file for more information."
+        with self.assertRaises(PluginException) as e:
+            self.action.run(self.params)
+        self.assertEqual(cause, e.exception.cause)
+        self.assertEqual(assistance, e.exception.assistance)
+        self.assertEqual(data, e.exception.data)
```
Comment (igorski-r7) [inline plugins/rapid7_insightvm/unit_test/test_tag_assets.py:59]: Same here
```suggestion
        with self.assertRaises(PluginException) as error:
            self.action.run(self.params)
        self.assertEqual(cause, error.exception.cause)
        self.assertEqual(assistance, error.exception.assistance)
        self.assertEqual(data, error.exception.data)
```

File: plugins/rapid7_insightvm/unit_test/test_tag_assets.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            "Verify your connection is pointing to your local console and not "
+            "`exposure-analytics.insight.rapid7.com` Ensure that the requested action does not cause a "
+            "conflict with the current state of the target resource."
+        )
+        data = "You cannot change the source of a tag."
+        with self.assertRaises(PluginException) as e:
+            self.action.run(self.params)
+        self.assertEqual(cause, e.exception.cause)
+        self.assertEqual(assistance, e.exception.assistance)
+        self.assertEqual(data, e.exception.data)
```
Comment (igorski-r7) [inline plugins/rapid7_insightvm/unit_test/test_tag_assets.py:74]:
```suggestion
        with self.assertRaises(PluginException) as error:
            self.action.run(self.params)
        self.assertEqual(cause, error.exception.cause)
        self.assertEqual(assistance, error.exception.assistance)
        self.assertEqual(data, error.exception.data)
```

File: plugins/rapid7_insightvm/unit_test/test_tag_assets.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        assistance = (
+            "Verify your connection is pointing to your local console and not "
+            "`exposure-analytics.insight.rapid7.com` Ensure that the requested action is permitted."
+        )
+        data = "You cannot create, edit or remove the tag 4 because it is a built-in tag."
+        with self.assertRaises(PluginException) as e:
+            self.action.run(self.params)
+        self.assertEqual(cause, e.exception.cause)
+        self.assertEqual(assistance, e.exception.assistance)
+        self.assertEqual(data, e.exception.data)
```
Comment (igorski-r7) [inline plugins/rapid7_insightvm/unit_test/test_tag_assets.py:88]:
```suggestion
        with self.assertRaises(PluginException) as error:
            self.action.run(self.params)
        self.assertEqual(cause, error.exception.cause)
        self.assertEqual(assistance, error.exception.assistance)
        self.assertEqual(data, error.exception.data)
```

**Review by igorski-r7 — CHANGES_REQUESTED** [review]
Looks great! 😃  Posted some comments there, but it's up to you. Let me know if you want to make those changes or not.

**Review by jkaluza-rapid7 — APPROVED** [review]

**Review by igorski-r7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Pull request merged**